schema_based_on_diagram.sql

```diff
@@ -0,0 +1,33 @@
+-- Active: 1693915950566@@127.0.0.1@7499
+CREATE DATABASE clinic;
+CREATE TABLE patients(id SERIAL PRIMARY KEY, name VARCHAR(40), date_of_birth DATE);
+CREATE INDEX ON patients (id);
+CREATE TABLE medical_histories(ID INT, admitted_at TIMESTAMP, patient_id INT, status VARCHAR(90),
+PRIMARY KEY(ID),
+FOREIGN KEY(patient_id) REFERENCES patients(id));
+CREATE INDEX ON medical_histories (ID);
+
+CREATE TABLE treatments(id INT PRIMARY KEY,  type VARCHAR(60), name VARCHAR(50));
+CREATE INDEX ON treatments (id);
+
+
+
+-- Now a talbe to join the id of treatment and medical history
+CREATE TABLE medical_treatment (
+    medical_history_id INT,
+    treatment_id INT,
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories(ID),
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+CREATE TABLE invoices(id SERIAL PRIMARY KEY, total_amount DECIMAL, generated_at TIMESTAMP, payed_at TIMESTAMP, medical_history_id INT,
+FOREIGN KEY(medical_history_id) REFERENCES medical_histories(ID));
+CREATE INDEX ON invoices (id);
+
+
+CREATE TABLE invoice_item(id SERIAL PRIMARY KEY, unit_price DECIMAL, quantity INT, total_price DECIMAL, invoice_id INT, treatment_id INT,
+FOREIGN KEY(invoice_id) REFERENCES invoices(id),
+FOREIGN KEY(treatment_id) REFERENCES treatments(id));
+CREATE INDEX ON invoice_item (id);
```

Comment:
- [ Required ] Good job implementing some of the changes from the previous reviewer, however, it would be nice if you created the indexes correctly. The `indexes` are supposed to be added to the `foreign keys` in a particular table and not on the `primary keys`. e.g on `invoice_items` the index should be on `treatment_id` and `invoice_id`, on table `invoices` the index should be on `medical_history_id` please see the examples below and rectify this.

## Example code
```SQL
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```